tests/rptest/services/rpk_producer.py

```diff
@@ -35,25 +36,34 @@ def __init__(self,
         self._output_line_count = 0
         self._partition = partition
         self._max_message_bytes = max_message_bytes
+        self._compression_type = compression_type
+        self._key_size = 16
 
         if produce_timeout is None:
             produce_timeout = 10
         self._produce_timeout = produce_timeout
 
+    @property
+    def approx_key_size(self):
```

Comment:
The "approx" part is a bit mysterious (and the test is assuming that these are lower bounds on the sizes, rather than approximations).

Kafka keys+values are binary arrays (should not be any encoding/null-handling involved), so the content we generate should have exactly the number of bytes as we are passing in.

Maybe the test is having to be approximate about this because of something happening in pandaproxy when the binary keys/values are munged into JSON strings?  If that's what's behind this, the approximation part should be specific to the test, and these methods should just be `key_size` and `msg_size`.

Reply:
IIRC, it basically does a minify on the JSON, so it should be predictable.